src/broadcast/mod.rs

```diff
@@ -39,12 +40,40 @@
 //! * A node that has received _2 f + 1_ `Ready`s **and** _N - 2 f_ `Echo`s with root hash `h`
 //! decodes and outputs the value, and then terminates.
 //!
-//! Only the first valid `Value` from the proposer, and the first valid `Echo` message from every
+//! We use a modified version of the algorithm to save on bandwith which provides the same
+//! security guarantees as the original. The main idea of the optimized algorithm is that in the
+//! optimisitic case, a node only needs _N - 2 f_ chunks to decode a value and every additional
+//! `Echo` message over that is wasteful.
+//! The modified algorithm introduces two new message types:
+//! * `CanDecode(h)` - Indicates node has enough shards to recover message with merkle root `h`.
+//! * `EchoHash(h)` - Indicates node can send an `Echo(p[i])` message upon request.
+//!
+//! Let `g` be the `fault_estimate` i.e. the estimate of number of faulty nodes in the network
+//! that we want to optimize for.
```

Comment:
We should state explicitly that the algorithm still works correctly if more than `g` are faulty.
Also, let's be consistent with the underscores (italic) and backticks (fixed width): I'd use underscores for formulas involving the single-letter identifiers (like _N_, _g_, _f_, …) and backticks only for identifiers that occur in the code or pseudocode (`Echo(p[i])`).